datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -888,10 +888,45 @@ fn estimate_join_cardinality(
             })
         }
 
-        JoinType::LeftSemi
-        | JoinType::RightSemi
-        | JoinType::LeftAnti
-        | JoinType::RightAnti => None,
+        JoinType::LeftSemi | JoinType::LeftAnti => {
+            let cardinality = estimate_semi_join_cardinality(
```

Comment:
it doesn't seem correct to me that the same calculation is used for both Semi and Anti joins (shouldn't they be the inverse of each other?)